grafana-plugin/src/state/rootBaseStore/index.ts

```diff
@@ -130,56 +128,63 @@ export class RootBaseStore {
   }
 
   setupPluginError(errorMsg: string) {
-    this.appLoading = false;
     this.initializationError = errorMsg;
   }
 
   /**
+   * This function is called in the background when the plugin is loaded.
+   * It will check the status of the plugin and
+   * rerender the screen with the appropriate message if the plugin is not setup correctly.
+   *
    * First check to see if the plugin has been provisioned (plugin's meta jsonData has an onCallApiUrl saved)
    * If not, tell the user they first need to configure/provision the plugin.
    *
    * Otherwise, get the plugin connection status from the OnCall API and check a few pre-conditions:
+   * - OnCall api should not be under maintenance
    * - plugin must be considered installed by the OnCall API
    * - token_ok must be true
    *   - This represents the status of the Grafana API token. It can be false in the event that either the token
    *   hasn't been created, or if the API token was revoked in Grafana.
    * - user must be not "anonymous" (this is determined by the plugin-proxy)
    * - the OnCall API must be currently allowing signup
-   * - the user must have an Admin role
-   * If these conditions are all met then trigger a data sync w/ the OnCall backend and poll its response
+   * - the user must have an Admin role and necessary permissions
    * Finally, try to load the current user from the OnCall backend
    */
   async setupPlugin(meta: OnCallAppPluginMeta) {
-    this.appLoading = true;
     this.initializationError = null;
     this.onCallApiUrl = meta.jsonData?.onCallApiUrl;
 
+    if (!FaroHelper.faro) {
+      FaroHelper.initializeFaro(this.onCallApiUrl);
+    }
```

Comment:
what's this for?

Reply:
[Tracing](https://grafana.com/oss/faro/)

Reply:
right. I just wasn't sure why this line was all of a sudden being added here but now I see it was removed [here](https://github.com/grafana/oncall/pull/2624/files#diff-c75a16456e6f502bc9eeb90ae7c381d244ded9046ac876186f5834734947b566L270). makes sense 👍